plugins/LilyPad/LilyPad.cpp

```diff
@@ -1201,14 +1206,19 @@ u8 CALLBACK PADpoll(u8 value)
         return query.response[++query.lastByte];
     }
 
-    int i;
     Pad *pad = &pads[query.port][query.slot];
+    int padtype = config.padConfigs[query.port][query.slot].type;
+
+    if (padtype != pad->previousType) {
```

Comment:
There are no other situations this currently occurs under, and any new situation would be handled elsewhere, so I've moved it to PADopen.